src/components/CytoscapeLayout/CytoscapeLayout.tsx

```diff
@@ -26,7 +28,8 @@ export default class CytoscapeLayout extends React.Component<CytoscapeLayoutProp
     console.log('Starting ServiceGraphPage for namespace ' + this.props.namespace);
 
     this.state = {
-      elements: []
+      elements: [],
+      isLoading: true
```

Comment:
Defaults to true? Shouldn't it only be set to true when we make the request?
Otherwise, the app first starts up it can have the empty state pattern.

Reply:
I don't understand. The first thing the app is going to do it to fetch the information from the server, so the default state when the page is first loaded is that its fetching data from the backend.

We could set it to be 'isLoading' when the server goes to do the fetch the first time. But then what should be displayed in this area for the split second before we try and fetch something from the server?

> Otherwise, the app first starts up it can have the empty state pattern.
I don't understand what you mean here. We first need to fetch the data from the backend, and then if its empty we show the empty state pattern. We shouldn't show the empty state pattern until we know if the data is actually empty or not.

Reply:
I think we should default `isLoading` to `false` and have `updateGraphElements` set `isLoading` to `true`.

Because `updateGraphElements` is also called when a property (the namespace) is changed, and I suppose it would be good to show the loading while loading more data.

Having it this way, we won't see an empty state because we are using `ComponentDidMount` to do the first `updateGraphElements` call (emphasis mine).

> Calling setState() in this method will trigger an extra rendering, but it will happen before the browser updates the screen. This guarantees that even though the render() will be called twice in this case, the **user won’t see the intermediate state**.



[1] https://reactjs.org/docs/react-component.html#componentdidmount

Reply:
This is the lifecycle:
1) It draws the component and creates the DOM first in `render`
2) Renders *all* child components
3) Only then does it do a `componentDidMount` and fetch

So there can be a longer than expected time between an initial state in the constructor and the actual fetch completing.

Reply:
I don't think we want to do the isLoading until we start the fetch.